### DISCUSSION TITLE
The server that the application is connecting to is unverified.

### DISCUSSION THREAD

**Discussion by vetaldnepr** [airsdk/Adobe-Runtime-Support#647] [Q&A]
Hi. In all my applications, when I start, I see a dialog with the same text.

AIR SDK version 32.0.0.116. After updating to version 33.1.1.98, the error remained.
The error is observed on all android devices.

**Comment by dayhiker16** [comment]
I have experienced the same thing. Multiple apps have this problem. So far, iOS apps are not effected. The code is identical for both Android and iOS. Used AIR SDK 32.1.1.98 for both. SSL cert on server is good for 3 years. In the third year now. Could new rules only allowing 2 year cert be the cause?

**Comment by FliplineStudios** [comment]
Is it a wildcard SSL certificate?  Seems like there's been a long-standing issue with wildcard certs with AIR and Android:  https://tracker.adobe.com/#/view/AIR-4198850

**Comment by dayhiker16** [comment]
Good point. I did know about this, but the problem only started today. 
The certs have not changed in two years. Why would the wildcard problem 
start now, and not previously? Hoping it was something else.

My certs do use a wild card cert for the Univ. of Oklahoma. My server is 
just one of thousands that use the global *.ou.edu cert. May need to buy a single cert, but was hoping to avoid this.

One other point, I do not use "StageWebView" that is documented in the
tracker.adobe.com page.

**Comment by ajwfrost** [comment]
This looks like we're receiving a certificate error report from the subsystem, so not necessarily something we'll be able to control..  but if it's possible to create a really simple test case for this, we can look into it to see what the actual reported error category is, in case there's something that can be set as 'acceptable' based on the certificate details..

thanks

**Comment by itlancer** [comment]
Seems it relevant to https://support.sectigo.com/articles/Knowledge/Sectigo-AddTrust-External-CA-Root-Expiring-May-30-2020 and very critical.

**Comment by vetaldnepr** [comment]
> 
> 
> This looks like we're receiving a certificate error report from the subsystem, so not necessarily something we'll be able to control.. but if it's possible to create a really simple test case for this, we can look into it to see what the actual reported error category is, in case there's something that can be set as 'acceptable' based on the certificate details..
> 
> thanks

Hi, ajwfrost.
In one of my projects I use the service: https://api.ipify.org, so i create a test case with it.
I will add a simple class and archive with the project, and dialog still exists!
AIR SDK version: 33.1.1.98.

Intellij IDEA project: [AIRTestApp.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4713201/AIRTestApp.zip)
or Main class:
```
package
{

    import flash.display.Sprite;
    import flash.events.ErrorEvent;
    import flash.events.Event;
    import flash.events.IOErrorEvent;
    import flash.events.SecurityErrorEvent;
    import flash.net.URLLoader;
    import flash.net.URLRequest;
    import flash.text.TextField;

    public class Main extends Sprite
    {
        public static var IP_API_URL:String = "https://api.ipify.org";

        private var _textField:TextField;

        public function Main()
        {
            _textField = new TextField();
            _textField.text = "Loading...";
            addChild(_textField);
            getRequest();
        }

        private function getRequest():void
        {
            var urlRequest:URLRequest = new URLRequest(IP_API_URL);
            var urlLoader:URLLoader = new URLLoader();
            urlLoader.addEventListener(Event.COMPLETE, onComplete_handler);
            urlLoader.addEventListener(IOErrorEvent.IO_ERROR, onError_handler);
            urlLoader.addEventListener(SecurityErrorEvent.SECURITY_ERROR, onError_handler);
            urlLoader.load(urlRequest);
        }

        private function onComplete_handler(event:Event):void
        {
            var urlLoader:URLLoader = event.target as URLLoader;
            urlLoader.removeEventListener(Event.COMPLETE, onComplete_handler);
            urlLoader.removeEventListener(IOErrorEvent.IO_ERROR, onError_handler);
            urlLoader.removeEventListener(SecurityErrorEvent.SECURITY_ERROR, onError_handler);
            _textField.text = "Success";
        }

        private function onError_handler(event:ErrorEvent):void
        {
            var urlLoader:URLLoader = event.target as URLLoader;
            urlLoader.removeEventListener(Event.COMPLETE, onComplete_handler);
            urlLoader.removeEventListener(IOErrorEvent.IO_ERROR, onError_handler);
            urlLoader.removeEventListener(SecurityErrorEvent.SECURITY_ERROR, onError_handler);
            _textField.text = "Error";
        }
    }
}

```

**Comment by dayhiker16** [comment]
Follow up. Thanks itLancer for the article link. That helped me search for possible reasons. Contacted my IT and it turns out that they had a "...Domain Verification for ou.edu expired on 5/30/2020 with our Certificate provider so we had to do another validation last week"  It should not have effected the app calling the server but the timing is suspect.  I installed new intermediate and root certificates they provided. That solved the problem (for now).

**Comment by zwetan** [comment]
Even with a valid SSL cert, it could be how the web server is configured
see [Mozilla Security/Server Side TLS](https://wiki.mozilla.org/Security/Server_Side_TLS)

Ideally you would want to test the domain via [Qualys SSL Server Test](https://www.ssllabs.com/ssltest/)

testing quickly `api.ipify.or` for example, under "Handshake Simulation" you will see which clients are supported or not

also the test will tell you if there are issues with your cert/server config

**Comment by jasonJamEther** [comment]
dayhiker16 I am running into the same issue with what appears to be the same ssl provider. Can you clarify your solution? You said you installed a new intermediate and root certificate. Did you install those on your server or added them somehow to your air app?

**Comment by dayhiker16** [comment]
Sorry to hear this. It was a weird issue, and still not sure why it 
happened. But yes, installing the new intermediate and root certificate 
on the server solved the issue. It was on the server, not in the apps. I 
did not change anything in the apps nor did I upload any new versions of 
the apps.

The server is a single, Windows 2008 (I know, it is old) server with 
just IIS web server running (not part of the apps). The apps call the 
server database (mySQL) for data.

Best of luck getting yours to work.

Kurt Gramoll, PhD
Professor
Aerospace and Mechanical Engineering
Univ. of Oklahoma
gramoll@ou.edu

**Comment by jasonJamEther** [comment]
dayhiker16 thanks for your help. We were able to solve it following your lead. We simply generated new crt files and replaced the old ones on the server. That fixed it.

**Comment by leossmith** [comment]
I am having the same issue in two client apps (they run on similar servers). Sadly their IT department said they wouldn't bother to fix this, as it doesn't affect their sites.

I was using their URLs to get Album Covers for music apps, and to solve this I had to use one of our company CDNs to get all their images from our system.

In native Android, there is a way to disable the popup window, so it might be a good idea to add this option in AIR too if other people have the same issue.
